engine/src/main/java/nl/inl/blacklab/searches/SearchCountFromResults.java

```diff
@@ -32,36 +26,18 @@
      */
     private final CountType type;
 
-    /**
-     * The (running or finished) result count.
-     * We can peek at this while it's running, or wait for executeInternal() to
-     * complete, returning the final results.
-     */
-    private ResultsStats resultCount;
-
-    /**
-     * Latch that marks the end of the search for this count.
-     */
-    private final CountDownLatch searchLatch;
-
-    /**
-     * Maximum time to wait for a search before giving up
-     */
-    private final int MAX_SEARCH_WAIT_IN_MS = 2000;
-
     public SearchCountFromResults(QueryInfo queryInfo, SearchForResults<T> source, CountType type) {
         super(queryInfo);
         this.source = source;
         this.type = type;
-        this.searchLatch = new CountDownLatch(1);
     }
 
     @Override
-    public ResultsStats executeInternal() throws InvalidQuery {
+    public ResultsStats executeInternal(Peekable<ResultsStats> progressReporter) throws InvalidQuery {
         // Start the search and construct the count object
-        resultCount = new ResultCount(source.executeNoQueue(), type);
-        // Mark the end of the search.
-        searchLatch.countDown();
+        ResultsStats resultCount = new ResultCount(source.executeNoQueue(), type);
+        if (progressReporter != null)
+            ((ResultsStatsDelegate) progressReporter.peek()).setRealStats(resultCount);
```

Comment:
Can the peek object be null as well here?, maybe both those checks can be done in the line above, like so:
```
if (progressReporter != null && progressReporter.peek() != null) {
}
```